kubernetes/structures_container.go

```diff
@@ -1028,14 +1028,3 @@ func expandContainerResourceRequirements(l []interface{}) (*v1.ResourceRequireme
 
 	return obj, nil
 }
-
-func flattenObjectRef(in *v1.ObjectReference) []interface{} {
-	att := make(map[string]interface{})
-	if in.Name != "" {
-		att["name"] = in.Name
-	}
-	if in.Namespace != "" {
-		att["namespace"] = in.Namespace
-	}
-	return []interface{}{att}
-}
```

Comment:
This was moved because it's only used in the volume schema.

Reply:
Technically `v1.ObjectReference` is used in a bunch of other resources (listed here: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.21/#objectreference-v1-core).

We might want to keep this and think about consolidating all those use-cases to re-use the same schema snippet and flatten / expand code (not in this PR, of course).

Reply:
Yeah, I did see that in the API docs, but in our codebase it's only used for the persistent volume schema, so I thought it made sense to keep it in that file until it's used more broadly. We may never use it outside of the persistent volume schema. But even if we do, the containers schema isn't quite the right place for it either, since it wouldn't be used in containers.